update(MQ): Message Queue improvements (#3209)
emqtt/emq-docs-cn: en_US/message-queue/message-queue-task.md

   - **Data Retention Period**: Specify how long messages should be retained in the queue. You can set the time unit (e.g., days).

>    - **Last Value Semantics**: This option is enabled by default. When enabled, a new message with the same queue key will overwrite any previous, unconsumed message with that key in the same queue. This ensures only the most recent message per key is retained.

Reviewer: ```suggestion
   - **Last Value Semantics**: This option is enabled by default. When enabled, a new message with the same queue key will overwrite any previous, unconsumed message with that key in the same queue. This ensures only the most recent message per key is retained. The default key is the ClientId of the message's publisher.
```

</details>

## Automatically Create Message Queue via Dashboard

Starting from EMQX 6.0.1, Message Queues can be automatically created when clients subscribe to a `$q/`-prefixed topic. This allows queues to be provisioned dynamically without manual setup.

> There are two types of auto-created queues: regular queue and last-value semantics queue. 

Reviewer: ```suggestion
The queues may be auto-created either as regular queues or last-value semantics queues. 
```

    gc_interval = 1h
    regular_queue_retention_period = 1d
    find_queue_retry_interval = 10s
    

>      limits {

Reviewer: Limits are part of individual queue configuration, not a part of global configuration.

### What happens when queues exceed capacity?

Message Queues in EMQX now support multiple types of capacity limits. If any of these limits are reached, EMQX will remove the oldest messages during garbage collection (GC) until the queue size returns within the configured bounds.

- **Time-based limit**: All queues are still subject to the configured *retention period*. Once messages exceed the retention period, they are no longer eligible for delivery and will be automatically purged during GC.

- **Size-based limits**: You can optionally configure per-shard limits on:

  - **Max number of messages** (`max_shard_message_count`)

>   - **Max total size of messages in bytes** (`max_shard_message_bytes`)

Reviewer: We probably should note, that the limit is
* per shard (and make a reference to the durable storage setting where the number of shards is configured)
* the size limit does not take replication factor in account, so the actual volume for a queue will be multiplied by the replication factor (also specified in durable storage settings).
